library/spdm_common_lib/libspdm_com_context_data.c

```diff
@@ -2220,6 +2220,42 @@ void libspdm_reset_context(void *context)
     #endif /* LIBSPDM_ENABLE_MSG_LOG */
 }
 
+/**
+ * Free the memory of contexts within the SPDM context.
+ * These are typically contexts whose memory has been allocated by the cryptography library.
+ * This function does not free the SPDM context itself.
+ *
+ * @param[in]  spdm_context         A pointer to the SPDM context.
+ *
+ */
+void libspdm_deinit_context(void *context)
+{
+#if !(LIBSPDM_RECORD_TRANSCRIPT_DATA_SUPPORT)
+    libspdm_context_t *spdm_context;
+    void *pubkey_context;
+    bool is_requester;
+
+    spdm_context = context;
+    is_requester = spdm_context->local_context.is_requester;
+    pubkey_context = spdm_context->connection_info.peer_used_cert_chain
+                     [spdm_context->connection_info.peer_used_cert_chain_slot_id].
```

Comment:
I think we need use a for-loop to free all 8 slots. (not only the used one)

Thinking about the case:
Step 1: requester gets all 8 slots (from 0 to 7).
Step 2: requester does challenge on slot 1.
Step 3: requester does key exchange on slot 2.